File: script/input.ts
export interface Root {
	sessions: Session[]
	speakers: Speaker[]
	session_types: SessionType[]
	rooms: Room[]
	tags: Tag[]
}

export interface Session {
	id: string
	type: string
	room: string
	broadcast?: string[]
	start: string
	end: string
	qa?: string
	slide?: string
	co_write?: string
	record: any
	live: any
	language: any
	uri?: string
	zh: Zh
	en: En
	speakers: string[]
	tags: string[]
}

export interface Zh {
	title: string
	description: string
}

export interface En {
	title: string
	description: string
}

export interface Speaker {
	id: string
	avatar: string
	zh: Zh2
	en: En2
}

export interface Zh2 {
	name: string
	bio: string
}

export interface En2 {
	name: string
	bio: string
}

export interface SessionType {
	id: string
	zh: Zh3
	en: En3
}

export interface Zh3 {
	name: string
	description: string
}

export interface En3 {
	name: string
	description: string
}

export interface Room {
	id: string
	zh: Zh4
	en: En4
}

export interface Zh4 {
	name: string
	description: string
}

export interface En4 {
	name: string
	description: string
}

export interface Tag {
	id: string
	zh: Zh5
	en: En5
}

export interface Zh5 {
	name: string
	description: string
}

export interface En5 {
	name: string
	description: string
}

File: backend/internal/refresh/input.go
package refresh

type Input struct {
	Sessions     []Session     `json:"sessions"`
	Speakers     []Speaker     `json:"speakers"`
	SessionTypes []SessionType `json:"session_types"`
	Rooms        []Room        `json:"rooms"`
	Tags         []Tag         `json:"tags"`
}

type Session struct {
	ID        string   `json:"id"`
	Type      string   `json:"type"`
	Room      string   `json:"room"`
	Broadcast []string `json:"broadcast"`
	Start     string   `json:"start"`
	End       string   `json:"end"`
	QA        string   `json:"qa,omitempty"`
	Slide     string   `json:"slide,omitempty"`
	CoWrite   string   `json:"co_write,omitempty"`
	Record    string   `json:"record"`
	Live      string   `json:"live"`
	URI       string   `json:"uri,omitempty"`
	Zh        struct {
		Title       string `json:"title"`
		Description string `json:"description"`
	} `json:"zh"`
	Speakers []string `json:"speakers"`
	Tags     []string `json:"tags"`
}

type Speaker struct {
	ID string `json:"id"`
	Zh struct {
		Name string `json:"name"`
	} `json:"zh"`
}

type Room struct {
	ID string `json:"id"`
	Zh struct {
		Name string `json:"name"`
	} `json:"zh"`
}

type SessionType struct {
	ID string `json:"id"`
	Zh struct {
		Name string `json:"name"`
	} `json:"zh"`
}

type Tag struct {
	ID string `json:"id"`
	Zh struct {
		Name string `json:"name"`
	} `json:"zh"`
}
